packages/api/src/controllers/stream.ts

```diff
@@ -1923,7 +1968,10 @@ app.patch("/:id/suspended", authorizer({}), async (req, res) => {
   const stream = await db.stream.get(id);
   if (
     !stream ||
-    (!req.user.admin && (stream.deleted || stream.userId !== req.user.id))
+    (!req.user.admin &&
+      (stream.deleted ||
+        stream.userId !== req.user.id ||
+        (stream.projectId ?? "") !== (req.project?.id ?? "")))
```

Comment:
Ok, so I'm gonna bring up this idea one last time, which is to create a helper function that checks access to a resource, so we don't have to keep repeating this whole logical chain everywhere that accesses some object in the DB. 

So I've implemented the helper function here https://github.com/livepeer/studio/commit/cb6f2d7e7b3ec22e408f9961ce885ac41d31d22f and use it in a couple different places just to show how it could be used. The idea would be just replacing the other access checks, seen on this PR, with a call to that function. It should hopefully make it easier to migrate the other data types to the new project-aware authorization flow as well, in the future.

Notice that I also added this which I think you forgot, to make sure we don't spam the db with queries for project objects that never change (we have it similar for users and api keys): https://github.com/livepeer/studio/commit/cb6f2d7e7b3ec22e408f9961ce885ac41d31d22f#diff-ce0ba37b6f5d180196d67f9e4b868e51df33660eac5213b4372701773a7b046bR65
I'd say that line is a required change, if you don't adopt the helper functions it'd be good to copy at least that part over.

> Disclaimer: I'm also only insisting because I don't recall getting a reply from you on the other comments I made while I think it'd make the code much less repetitive and error-prone (like the inverted checks on `hasAccess` below). I feel the suggestion could feel a little daunting without the TypeScript experience to refactor some stuff around as well, so I thought providing you with a commit head start could be helpful. I wrote that commit in 5 minutes and no problem if you prefer to stick with your current implementation tho.